server/controllers/admin/suppliers.js

```diff
@@ -124,5 +124,48 @@ exports.update = function update(req, res, next) {
   })
   .catch(next)
   .done();
-};
+}
+
+// GET /SUPPLIER/FILTER
+
+function filter(req, res, next) { 
```

Comment:
A better name for this endpoint is `search()`.  We use `search()` in both the `patients` and the `sales` controllers for this functionality.